### ISSUE TITLE
Codegen does not visit P in Punctuated<T, P>

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/syn#291]
Only the T get visited.

```toml
[dependencies]
syn = { version = "0.14", features = ["visit", "extra-traits"] }
proc-macro2 = "0.4"
```

```rust
extern crate syn;
use syn::visit::Visit;

extern crate proc_macro2;
use proc_macro2::Span;

struct SpanVisitor;

impl<'ast> Visit<'ast> for SpanVisitor {
    fn visit_span(&mut self, span: &'ast Span) {
        println!("{:?}", span);
    }
}

fn main() {
    let path = syn::parse_str("syn::visit::Visitor").unwrap();
    println!("{:#?}", path);
    SpanVisitor.visit_path(&path); // only 3 spans
}
```

**Comment by mystor** [comment]
Oops. I think this design decision was made when we didn't visit Span fields in structs so I figured I didn't ever need to traverse the P fields of a Punctuated.

The easiest way would probably be to treat it like a vector of tuples.

**Comment by dtolnay** [comment]
Fixed in d81f91f5bd77b9bd1d24861af1e6ae18ddc5e997 + b26ec8783949cccefcaddcf51ece2e401035b71e.
